code/game/machinery/dye_generator.dm

```diff
@@ -2,8 +2,9 @@
 	name = "Dye Generator"
 	icon = 'icons/obj/vending.dmi'
 	icon_state = "barbervend"
-	density = 1
-	anchored = 1
+	density = TRUE
+	anchored = TRUE
+	integrity_failure = 100
```

Comment:
Undocumented change

Reply:
Updated the description. Basically, it'd just disappear and wouldn't switch to a broken state due to defaulting to 0. This brings it in line with vendors.